core/tx_pool.go

```diff
@@ -792,22 +889,28 @@ func (pool *TxPool) addTxs(txs []*types.Transaction, local, sync bool) []error {
 		types.Sender(pool.signer, tx)
 	}
 	// Process all the new transaction and merge any errors into the original slice
-	pool.mu.Lock()
-	newErrs, dirtyAddrs := pool.addTxsLocked(news, local)
-	pool.mu.Unlock()
-
+	insertOp := &txsInsert{
+		txs:   news,
+		local: local,
+		sync:  sync,
+		done:  make(chan struct{}),
+	}
+	select {
+	case pool.insertTxsCh <- insertOp:
+		<-insertOp.done
+	case <-pool.shutdownCh:
+		return pool.closedErrs(len(txs))
+	}
+	if insertOp.errs == nil {
+		return pool.closedErrs(len(txs))
+	}
 	var nilSlot = 0
-	for _, err := range newErrs {
+	for _, err := range insertOp.errs {
 		for errs[nilSlot] != nil {
 			nilSlot++
 		}
 		errs[nilSlot] = err
 	}
```

Comment:
It looks to me like you just drop the `sync` flag on the floor...? The sync flag means that the caller don't want this method to return until the txs have been properly handled. I think with this fix, we just pass them off to some other thread and returns, regardless of `sync` or not... ?

Reply:
Nope, the sync is handled in another loop. If sync is specified, the `done` channel will be closed after all internal processes are done

Reply:
https://github.com/ethereum/go-ethereum/pull/21171/files#diff-8fecce9bb4c486ebc22226cf681416e2R455 Here

Reply:
I have a question here, I would like to ask.
If local transactions are being sent all the time and remote transactions are being received all the time, while maintaining the priority to insert local transactions, will remote transactions continue to skyrocket in memory?

Reply:
In the real network, we can receive transactions from local or p2p network. If there are too many remote transactions accumulated, then it will block the network connection to send more data since it has to wait that the transactions are actually processed(we wait the done channel). So it will give the back pressure to the transaction generators.

What if we have countless transaction generators? In the real use case, we only have limited peer connections. So it's fine. Also, we have the assumption that local users won't ddos the node.